code/__DEFINES/spatial_gridmap.dm

```diff
@@ -11,6 +12,11 @@
 #define SPATIAL_GRID_CONTENTS_TYPE_HEARING RECURSIVE_CONTENTS_HEARING_SENSITIVE
 ///every movable that has a client in it is stored in this channel
 #define SPATIAL_GRID_CONTENTS_TYPE_CLIENTS RECURSIVE_CONTENTS_CLIENT_MOBS
+///all atmos machines are stored in this channel (I'm sorry kyler)
+#define SPATIAL_GRID_CONTENTS_TYPE_ATMOS "spatial_grid_contents_type_atmos"
 
 ///whether movable is itself or containing something which should be in one of the spatial grid channels.
-#define HAS_SPATIAL_GRID_CONTENTS(movable) (movable.important_recursive_contents && (movable.important_recursive_contents[RECURSIVE_CONTENTS_HEARING_SENSITIVE] || movable.important_recursive_contents[RECURSIVE_CONTENTS_CLIENT_MOBS]))
+#define HAS_SPATIAL_GRID_CONTENTS(movable) ((\
+	movable.important_recursive_contents && (\
+	movable.important_recursive_contents[RECURSIVE_CONTENTS_HEARING_SENSITIVE] || movable.important_recursive_contents[RECURSIVE_CONTENTS_CLIENT_MOBS]))\
+	|| movable.flags_1 & SPATIAL_GRID_MANAGED_1)
```

Comment:
i dont like this. itd be best for code quality reasons if spatial grid contents were made into an element hooked onto the moved signal but >adding _SendSignal() overhead to movement just to do no work more than 16/17ths of the time

ideally there would be one check for whether something is in the grid. question is whats the best way to do that. obviously forcing everything through important_recursive_contents isnt ideal, but the things that are there already on master need to continue working inside nested locs. perhaps the best way to check for this is a trait that is propagated along with important_recursive_contents for the contents type that do that. ill have to think about this

Reply:
It could totally be a trait, the memory cost would be a thing, but it wouldn't be "that" bad.
If you wanted to really hack it you could implement trait like behavior manually, with some other single var way of indicating a thing being inherently interested in being managed by the grid

Reply:
34b7b261467383136cc92f65f35f4642e9ef43f9